pkg/network/deviceinfo/deviceinfo_test.go

```diff
@@ -0,0 +1,208 @@
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* Copyright 2022 Red Hat, Inc.
+*
+ */
+package deviceinfo_test
+
+import (
+	"fmt"
+
+	"kubevirt.io/kubevirt/pkg/network/deviceinfo"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	virtv1 "kubevirt.io/api/core/v1"
+)
+
+var _ = Describe("DeviceInfo", func() {
+
+	const (
+		booHashedIfaceName  = "pod6446d58d6df"
+		fooHashedIfaceName  = "pod2c26b46b68f"
+		fooOrdinalIfaceName = "net1"
+		booOrdinalIfaceName = "net2"
+	)
+
+	networkStatusWithMixedNetworksFmt := `
+	[
+	{
+	"name": "kindnet",
+	"interface": "eth0",
+	"ips": [
+	  "10.244.1.9"
+	],
+	"mac": "3a:7e:42:fa:37:c6",
+	"default": true,
+	"dns": {}
+	},
+	{
+	"name": "default/nad1",
+	"interface": "%s",
+	"mac": "8a:37:d9:e7:0f:18",
+	"dns": {}
+	},
+	{
+	"name": "default/nad2",
+	"interface": "%s",
+	"dns": {},
+	"device-info": {
+	  "type": "pci",
+	  "version": "1.0.0",
+	  "pci": {
+	    "pci-address": "0000:65:00.2"
+	  }
+	}
+	}
+	]`
+	networkStatusWithNoDeviceInfoFmt := `
+[
+{
+ "name": "kindnet",
+ "interface": "eth0",
+ "ips": [
+   "10.244.2.131"
+ ],
+ "mac": "82:cf:7c:98:43:7e",
+ "default": true,
+ "dns": {}
+},
+{
+ "name": "default/nad1",
+ "interface": "%s",
+ "dns": {},
+}
+]`
+	networkStatusWithPrimaryInterfaceOnlyFmt := `
+[
+{
+ "name": "kindnet",
+ "interface": "eth0",
+ "ips": [
+   "10.244.2.131"
+ ],
+ "mac": "82:cf:7c:98:43:7e",
+ "default": true,
+ "dns": {}
+}
+]`
+	DescribeTable("should prepare empty network device info annotation",
+		func(networkList []virtv1.Network, interfaceList []virtv1.Interface, networkStatusAnnotationValue string) {
+			networkDeviceInfoAnnotationValue := deviceinfo.CreateNetworkDeviceInfoAnnotationValue(networkList, interfaceList, networkStatusAnnotationValue)
+			Expect(networkDeviceInfoAnnotationValue).To(Equal(""))
+		},
+		Entry("when networkStatusAnnotation deson't contain the interface",
+			[]virtv1.Network{
+				newMultusNetwork("foo", "default/nad1"),
+			},
+			[]virtv1.Interface{
+				newBindingPluginInterface("foo"),
+			},
+			fmt.Sprintf(networkStatusWithPrimaryInterfaceOnlyFmt),
+		),
+		Entry("when networkStatusAnnotation has no interfaces with device info",
+			[]virtv1.Network{
+				newMultusNetwork("foo", "default/nad1"),
+			},
+			[]virtv1.Interface{
+				newBindingPluginInterface("foo"),
+			},
+			fmt.Sprintf(networkStatusWithNoDeviceInfoFmt, fooHashedIfaceName, booHashedIfaceName),
+		),
+		Entry("when pod's networkStatus Annotation does not exist",
+			[]virtv1.Network{newMultusNetwork("foo", "default/nad1")},
+			[]virtv1.Interface{newBindingPluginInterface("foo")},
+			"",
+		),
+	)
+
+	DescribeTable("should prepare non empty network device info annotation",
```

Comment:
It seems that only the expected annotation varies between the tests in this table, specifically the pod interface names, hashed vs. ordinal.
Please consider simplifying it by passing only the expected annotation.

Reply:
done.